lib/developer_portal/app/controllers/developer_portal/admin/account/invoices_controller.rb

```diff
@@ -31,4 +37,20 @@ def authorize_finance
   def find_provider
     @provider = site_account
   end
+
+  delegate :payment_gateway_options, to: :site_account
+
+  def payment_intent
+    @payment_intent ||= @invoice.payment_intents.latest_pending.first
+  end
+
+  def api_key
+    payment_gateway_options[:login]
+  end
+
+  def fetch_stripe_client_secret
+    return unless payment_intent
```

Comment:
Maybe here we should raise a 404 if the payment intent is missing.

Related to https://github.com/3scale/porta/pull/2350/files?file-filters%5B%5D=.liquid&file-filters%5B%5D=.rb#r568718935.